src/data/markdown/docs/02 javascript api/07 k6-experimental/01 browser/02 BrowserContext/waitForEvent.md

```diff
@@ -1,33 +1,71 @@
 ---
 title: 'waitForEvent(event[, optionsOrPredicate])'
-excerpt: 'Waits for event to fire and passes its value into the predicate function.'
+excerpt: 'Waits for event to fire and returns its value.'
 ---
 
-<Blockquote mod="attention">
+Waits for the event to fire and returns its value. If a predicate function has been set it will pass the value to the predicate function, which must return `true` for the promise to resolve.
 
-This method is a work in progress.
-It requires async functionality and returning a `Promise` to be useful in scripts.
-Refer to <a href="https://github.com/grafana/xk6-browser/issues/447">#447</a> for details.
+<TableWithNestedRows>
 
-Consider using the sync methods `Page.waitForNavigation()` and `Page.waitForSelector()` instead.
+| Parameter                    | Type             | Default | Description                                                                                                                                           |
+|------------------------------|------------------|---------|-------------------------------------------------------------------------------------------------------------------------------------------------------|
+| event                        | string           | `null`  | Name of event to wait for. Currently the `'page'` event is the only one that is supported.                                                            |
+| optionsOrPredicate           | function\|object | `null`  | Optional. If it's a function, the `'page'` event data will be passed to it and it must return `true` to continue.                                     |
+| optionsOrPredicate.predicate | function         | `null`  | Optional. Function that will be called when the `'page'` event is emitted. The event data will be passed to it and it must return `true` to continue. |
+| optionsOrPredicate.timeout   | number           | `30000` | Optional. Maximum time to wait in milliseconds.                                                                                                       |
 
-</Blockquote>
+</TableWithNestedRows>
 
-Waits for the event to fire and passes its value into the predicate function. Returns the event data value when the predicate returns `true`.
+### Returns
 
-<TableWithNestedRows>
+| Type            | Description                                                                                           |
+|-----------------|-------------------------------------------------------------------------------------------------------|
+| Promise<Object> | At the moment a [Page](/javascript-api/k6-experimental/browser/page/) object is the only return value |
```

Comment:
I believe we have to escape these `<` `>`. See related CI error [here](https://github.com/grafana/k6-docs/actions/runs/6669691353/job/18128069484?pr=1384#step:10:49).

Reply:
Thanks for that, fixed in a fixup e49bca0afaca619b1466fa59b84abd28bf800361